pytorch_lightning/profiler/pytorch.py

```diff
@@ -116,7 +116,8 @@ def pre_step(self, current_action: str) -> None:
         self._current_action = current_action
 
     def reset(self):
-        self._num_optimizer_step_and_closure = 0
+        # handle properly `fast_dev_run`. PyTorch Profiler will fail otherwise.
+        self._num_optimizer_step_and_closure = -1
```

Comment:
if `fast_dev_run>1`, this would be wrong?

Reply:
Yes. But don t have a solution so far. It seems closing the Profiler during warmup doesn t play nice.

Reply:
but why -1? are you sure it's not the sanity val check?

Reply:
Removed !